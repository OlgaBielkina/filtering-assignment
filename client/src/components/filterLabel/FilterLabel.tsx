import React, { useCallback } from 'react';
import classnames from 'classnames';
import styles from './FilterLabel.module.scss';

export interface Props {
    label: string;
    nodeId: string;
    onSelect: (nodeId: string, isSelected: boolean) => void;
}

export const FilterLabel: React.FC<Props> = ({ onSelect, label, nodeId }) => {
    const clickHandler = useCallback(() => {
        onSelect(nodeId, false);
    }, [nodeId, onSelect]);

    return (
        <span onClick={clickHandler} className={styles.root}>
            <span className={classnames('material-icons', styles.closeIcon)}>close</span>
            <span className={styles.label}>{label}</span>
        </span>
    );
};
