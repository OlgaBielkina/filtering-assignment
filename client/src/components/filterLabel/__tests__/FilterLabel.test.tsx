import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { FilterLabel, Props } from '../FilterLabel';

describe('FilterLabel', () => {
    const props: Props = {
        nodeId: '1',
        label: 'test name',
        onSelect: jest.fn(),
    };

    it('should render correctly', () => {
        const { asFragment } = render(<FilterLabel {...props} />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('should call onSelect callback when pressing a label', () => {
        const { getByText } = render(<FilterLabel {...props} />);

        fireEvent.click(getByText(props.label));

        expect(props.onSelect).toBeCalledTimes(1);
    });
});
