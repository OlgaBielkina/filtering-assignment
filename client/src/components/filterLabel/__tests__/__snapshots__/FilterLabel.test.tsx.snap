// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLabel should render correctly 1`] = `
<DocumentFragment>
  <span
    class="root"
  >
    <span
      class="material-icons closeIcon"
    >
      close
    </span>
    <span
      class="label"
    >
      test name
    </span>
  </span>
</DocumentFragment>
`;
